packages/testing/config/jest/api/jest.setup.js

```diff
@@ -31,12 +57,27 @@ const seedScenario = async (scenario) => {
 }
 
 const teardown = async () => {
-  const prismaModelNames = (await getSchemaDefinitions()).datamodel.models.map(
-    (m) => m.dbName || m.name
-  )
+  for (const modelName of teardownOrder) {
+    try {
+      await db.$executeRawUnsafe(`DELETE from "${modelName}"`)
+    } catch (e) {
+      if (e.message.match(/FOREIGN KEY constraint failed/)) {
+        const index = teardownOrder.indexOf(modelName)
+        teardownOrder[index] = null
+        teardownOrder.push(modelName)
+      } else {
+        throw e
+      }
+    }
+  }
+
+  // remove nulls
+  teardownOrder = teardownOrder.filter((val) => val)
 
-  for (const model of prismaModelNames) {
-    await db.$executeRawUnsafe(`DELETE from "${model}"`)
+  // if the order of delete changed, write out the cached file again
+  if (JSON.stringify(teardownOrder) !== JSON.stringify(originalTeardownOrder)) {
```

Comment:
This code will be a lot easier to parse if we read the details just above the for loop, otherwise we'll be JSON parsing for each model